lib/net.c

```diff
@@ -720,10 +720,62 @@ get_if_back(struct net_config *net_conf)
 	ETH_RSS_NONFRAG_IPV6_OTHER | \
 	ETH_RSS_IPV6_EX)
 
+static int
+check_port_rss_key_update(struct gatekeeper_if *iface, uint16_t port_id,
+	bool *rss_key_updatable, struct rte_eth_rss_conf *rss_conf)
+{
+	struct rte_eth_rss_conf cur_rss_conf;
+	int ret = rte_eth_dev_rss_hash_update(port_id, rss_conf);
+	if (ret == 0)
+		return ret;
+
+	ret = rte_eth_dev_rss_hash_conf_get(port_id, &cur_rss_conf);
+	if (ret < 0) {
+		G_LOG(WARNING,
+			"net: the %s interface does not support a critical hardware capability: update and get the RSS hash configuration\n",
+			iface->name);
+		return ret;
+	}
+
+	/*
+	 * At least two ports with different RSS hash keys
+	 * cannot be updated.
+	 */
+	if (!*rss_key_updatable &&
+			(cur_rss_conf.rss_key == NULL ||
+			cur_rss_conf.rss_key_len !=
+			rss_conf->rss_key_len ||
+			memcmp(rss_conf->rss_key,
+			cur_rss_conf.rss_key,
+			rss_conf->rss_key_len) != 0))
+		return -EINVAL;
+
+	if (cur_rss_conf.rss_key_len > GATEKEEPER_RSS_KEY_LEN)
+		return -EINVAL;
+
+	/* Update the default RSS hash key. */
+	memset(default_rss_key, 0, RTE_DIM(default_rss_key));
```

Comment:
`memset()` expects the size in bytes, so use `sizeof(default_rss_key)` instead of `RTE_DIM(default_rss_key)`. Notice that the type of the array `default_rss_key` is `uint8_t`, so both expressions are equivalent. But given the nature of `memset()`, `sizeof()` is preferable because it doesn't depend on the type.